src/JsonApiDotNetCore.MongoDb/MongoDbRepository.cs

```diff
@@ -0,0 +1,230 @@
+            }
+
+            return Collection.InsertOneAsync(resourceForDatabase, new InsertOneOptions(), cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public virtual async Task<TResource> GetForUpdateAsync(QueryLayer queryLayer, CancellationToken cancellationToken)
+        {
+            var resources = await GetAsync(queryLayer, cancellationToken);
+            return resources.FirstOrDefault();
+        }
+
+        /// <inheritdoc />
+        public virtual async Task UpdateAsync(TResource resourceFromRequest, TResource resourceFromDatabase, CancellationToken cancellationToken)
+        {
+            if (resourceFromRequest == null) throw new ArgumentNullException(nameof(resourceFromRequest));
+            if (resourceFromDatabase == null) throw new ArgumentNullException(nameof(resourceFromDatabase));
+            
+            foreach (var attr in _targetedFields.Attributes)
+                attr.SetValue(resourceFromDatabase, attr.GetValue(resourceFromRequest));
+
+            await Collection.ReplaceOneAsync(
+                Builders<TResource>.Filter.Eq(e => e.Id, resourceFromDatabase.Id),
+                resourceFromDatabase,
+                new ReplaceOptions(),
+                cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public virtual async Task DeleteAsync(TId id, CancellationToken cancellationToken)
+        {
+            var result = await Collection.DeleteOneAsync(
+                Builders<TResource>.Filter.Eq(e => e.Id, id),
+                new DeleteOptions(),
+                cancellationToken);
+
+            if (!result.IsAcknowledged)
+            {
+                throw new DataStoreUpdateException(new Exception("Delete operation was not acknowledged by MongoDB"));
+            }
+
+            if (result.DeletedCount == 0)
+            {
+                throw new DataStoreUpdateException(new Exception($"No documents were deleted. The id that was being sought was: {id}"));
+            }
+        }
+
+        /// <inheritdoc />
+        public virtual Task SetRelationshipAsync(TResource primaryResource, object secondaryResourceIds, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc />
+        public virtual Task AddToToManyRelationshipAsync(TId primaryId, ISet<IIdentifiable> secondaryResourceIds, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc />
+        public virtual Task RemoveFromToManyRelationshipAsync(TResource primaryResource,
+            ISet<IIdentifiable> secondaryResourceIds,
+            CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+    }
+    
+    /// <summary>
+    /// Implements the foundational repository implementation that uses MongoDB.
```

Comment:
It looks like you took this comment from the main project, but it is wrong there. They should be the same on both types:
```
Implements the foundational Repository layer in the JsonApiDotNetCore architecture that uses MongoDB.
```